lib/que/adapters/active_record_with_lock.rb

```diff
@@ -0,0 +1,62 @@
+# frozen_string_literal: true
+
+# https://github.com/que-rb/que/blob/80d6067861a41766c3adb7e29b230ce93d94c8a4/lib/que/active_job/extensions.rb
+module Que
+    module Adapters
+      class ActiveRecordWithLock < Que::Adapters::ActiveRecord
+        attr_accessor :job_connection_pool, :lock_connection
+        def initialize(job_connection_pool:, lock_connection:)
+            @job_connection_pool = job_connection_pool
+            @lock_connection = lock_connection
+            super
+        end
+
+        def checkout_activerecord_adapter(&block)
+            @job_connection_pool.with_connection(&block)
+        end
+
+        def lock_database_connection
+            Thread.current[:db_connection] ||= @lock_connection.connection
```

Comment:
```suggestion
            Thread.current[:db_connection] ||= @lock_record.connection
```
does this read better?